src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -549,6 +561,25 @@ public OpenIdConnectHandler(HttpClient backchannel)
                     return null;
                 }
                 ticket = authorizationCodeReceivedContext.AuthenticationTicket;
+
```

Comment:
`message = authenticationValidatedContext.ProtocolMessage;` since you're using it after the event.